protocol/localstatequery/localstatequery.go

```diff
@@ -121,6 +129,13 @@ type Config struct {
 	QueryTimeout   time.Duration
 }
 
+// Helper types
+type AcquirePoint struct {
+	Point     *common.Point
+	Volatile  bool
+	Immutable bool
```

Comment:
Can it ever be neither or both? Are these completely independent from each other?

Reply:
It can only be one of them at a time. It might make a little more sense to have a "type" field to make it a bit more explicit which one it is, or even separate types for each with an interface type

Reply:
I switch this to 4 separate types, one for each acquire target type, and an interface that covers all of them for convenience.